akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -289,7 +289,10 @@ private[http] object HttpServerBluePrint {
         def onPush(): Unit = {
           openTimeouts.head.clear()
           openTimeouts = openTimeouts.tail
-          push(responseOut, grab(responseIn))
+
+          // it may have been completed by a timeout already
```

Comment:
cancel this port is happening actually (thus the `isAvailable` == false). 
I did not try `completeStage()` instead `complete(responseOut)` actually... need to think if it changes semantics or not hm